src/index.ts

```diff
@@ -1,226 +1,180 @@
-}
-
-export class CachedClient implements ICachedHttpClient {
-    private _httpClient: IHttpClient;
-    private _logger: Logger;
-    private _cache: Cache;
-    constructor(httpClient: HttpClient, logger: Logger, cache: Cache) {
-        this._httpClient = httpClient;
-        this._logger = logger;
-        this._cache = cache;
-    }
-
-    private async _get<T>(route: string, noCache: boolean): Promise<T> {
-
-        if (!noCache) {
-            const cached = this._cache.get<T>(route);
-            if (cached) {
-                this._logger && this._logger.info(`returning cached result for route "${route}"`);
-                return cached;
-            }
-        }
-        const result = await this._httpClient.get<T>(route);
-        this._cache.set(route, result);
-        return result;
-    }
-
-    public async get<T>(route: string): Promise<T> { return this._get(route, false); }
-    public async getNoCache<T>(route: string): Promise<T> { return this._get(route, true); }
-
-    public async post<T>(route: string, payload?: object): Promise<T> {
-        return await this._httpClient.post<T>(route, payload);
-    }
-
-    public async put<T>(route: string, payload?: object): Promise<T> {
-        return await this._httpClient.put<T>(route, payload);
-    }
-
-    public async patch<T>(route: string, payload?: object): Promise<T> {
-        return await this._httpClient.patch<T>(route, payload);
-    }
-
-    public async delete(route: string): Promise<void> {
-        return await this._httpClient.delete(route);
-    }
+/**
+ * HttpClient to make network requests
+ */
+export class HttpClient implements IHttpClient {
+  private _got: Got;
+  private _logger: Logger;
+  private _prefixUrl: string;
+
+  constructor(config: HttpClientConfiguration) {
+    const options: Options = { ...config };
+
+    const addToHeader = (header: Record<string, string>) => {
+      options.headers ? Object.assign(options.headers, header) : (options.headers = header);
+    };
+
+    // default authorization header
+    if (options.context?.basicAuthUserName && options.context?.basicAuthPassword) {
+      addToHeader({
+        Authorization: `Basic ${Buffer.from(
+          options.context.basicAuthUserName + ':' + options.context.basicAuthPassword
+        ).toString('base64')}`
+      });
+    } else if (options.context?.bearerToken) {
+      addToHeader({ Authorization: 'Bearer ' + options.context.bearer });
+    }
+
+    // ininitialize logger
+    if (config?.context?.logger) {
+      this._logger = config.context.logger;
+      delete config.context.logger;
+    } else {
+      this._logger = console;
+    }
+
+    //save prefixUrl
+    this._prefixUrl = config.prefixUrl;
+
+    // initialize got and extend it with default options
+    this._got = got.extend({ ...options, retry: 0 });
+  }
+
+  private _request(path: string, options: RequestOptions): CancelableRequest<Response<string>> {
+    const { abortSignal, ...gotOptions } = options || {};
```

Comment:
Safety: These day it's better to use `??` over `||` in the majority of cases (since it's stricter and won't care about coercion).